packages/x-components/src/x-modules/search/components/banners-list.vue

```diff
@@ -0,0 +1,256 @@
+<template>
+  <NoElement>
+    <!--
+      @slot Customized Banners List layout.
+          @binding {banners} banners - Banners to render.
+          @binding {gridItems} gridItems - `Results` and `injectedGridItems`.
+          @binding {animation} animation - Animation to animate the elements.
+    -->
+    <slot v-bind="{ banners, gridItems, animation }">
+      <component
+        :is="animation"
+        v-if="banners.length"
+        tag="ul"
+        class="x-list x-banners-list"
+        data-test="banners-list"
+      >
+        <li
+          v-for="banner in banners"
+          :key="banner.id"
+          class="x-banners-list__item"
+          data-test="banners-list-item"
+        >
+          <!--
+            @slot Customized Banners List banner.
+                @binding {banner} banner - Banner data
+          -->
+          <slot :banner="banner" name="banner">{{ banner.title }}</slot>
+        </li>
+      </component>
+    </slot>
+  </NoElement>
+</template>
+
+<script lang="ts">
+  import { Banner } from '@empathyco/x-types';
+  import Vue from 'vue';
+  import { Component, Prop } from 'vue-property-decorator';
+  import { State } from '../../../components/decorators/store.decorators';
+  import { NoElement } from '../../../components/no-element';
+  import { xComponentMixin } from '../../../components/x-component.mixin';
+  import { searchXModule } from '../x-module';
+  import { XInject, XProvide } from '../../../components/decorators/injection.decorators';
+  import { GridItem } from '../../../utils/types';
+  /**
+   * It renders a list of banners from {@link SearchState.banners} by default.
+   *
+   * The component provides a default slot which wraps the whole component with the `banners` bound
+   * and the `gridItems` which also contains the injected grid items from an ancestor.
+   *
+   * It also provides the slot banner to customize the item, which is within the default slot, with
+   * the result bound.
+   *
+   * @public
+   */
+  @Component({
+    components: {
+      NoElement
+    },
+    mixins: [xComponentMixin(searchXModule)]
+  })
+  export default class BannersList extends Vue {
+    /**
+     * The results to render.
+     *
+     * @public
+     */
+    @State('search', 'banners')
+    public banners!: Banner[];
+
+    /**
+     * Animation component that will be used to animate the promoteds.
+     *
+     * @public
+     */
+    @Prop({ default: 'ul' })
+    protected animation!: Vue | string;
+
+    /**
+     * It injects gridItems provided by an ancestor as injectedGridItems.
+     *
+     * @internal
+     */
+    @XInject('gridItems', [] as GridItem[])
```

Comment:
Extract `gridItems` to a constant like this, and use it everywhere.
```
const GridItemsKey: XInjectKey<GridItem[]> = 'gridItems';
```

Reply:
Done